### PR TITLE
Authentication with JWT

### PR DESCRIPTION
**Pull request by michalwski**
This PR is just a fix for #977

### CODE REVIEW

File: apps/ejabberd/src/ejabberd_auth_jwt.erl [hunk 1]
```diff
@@ -0,0 +1,215 @@
+%%%----------------------------------------------------------------------
+%%% File    : ejabberd_auth_internal.erl
```
Comment (fenek) [inline apps/ejabberd/src/ejabberd_auth_jwt.erl]: Invalid file name in comment

File: apps/ejabberd/test/auth_jwt_SUITE.erl [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+get_password(_Config) ->
+    <<>> = ejabberd_auth_jwt:get_password_s(<<"anakin">>, ?DOMAIN1).
+
+is_user_exists(_Config) ->
+    true = ejabberd_auth_jwt:does_user_exist(<<"madhatter">>, ?DOMAIN1).
+
+% remove_user/2,3
+remove_user(_Config) ->
+    ok = ejabberd_auth_jwt:remove_user(<<"toremove3">>, ?DOMAIN1),
+    {error, not_allowed} = ejabberd_auth_jwt:remove_user(<<"toremove3">>, ?DOMAIN1, <<"wrongpass">>).
```
Comment (gadgetci) [inline apps/ejabberd/test/auth_jwt_SUITE.erl]: According to **Elvis**:
> Line 98 is too long:     {error, not_allowed} = ejabberd_auth_jwt:remove_user(<<"toremove3">>, ?DOMAIN1, <<"wrongpass">>)..

File: apps/ejabberd/test/auth_jwt_SUITE.erl [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    false = ejabberd_auth_jwt:check_password(?USERNAME, ?DOMAIN1, generate_token(60)).
+
+check_password_fails_for_correct_token_but_wrong_username(_C) ->
+    false = ejabberd_auth_jwt:check_password(<<"alice">>, ?DOMAIN1, generate_token(0)).
+
+set_password(_Config) ->
+    {error, not_allowed} = ejabberd_auth_jwt:set_password(<<"alice">>, ?DOMAIN1, <<"mialakota">>).
+
+try_register(_Config) ->
+    {error, not_allowed} = ejabberd_auth_jwt:try_register(<<"nonexistent">>, ?DOMAIN1, <<"newpass">>).
```
Comment (gadgetci) [inline apps/ejabberd/test/auth_jwt_SUITE.erl]: According to **Elvis**:
> Line 86 is too long:     {error, not_allowed} = ejabberd_auth_jwt:try_register(<<"nonexistent">>, ?DOMAIN1, <<"newpass">>)..

File: apps/ejabberd/src/ejabberd_auth_jwt.erl [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+-spec get_password_s(LUser :: ejabberd:luser(),
+                     LServer :: ejabberd:lserver()) -> binary().
+get_password_s(_LUser, _LServer) ->
+    <<"">>.
+
+-spec does_user_exist(LUser :: ejabberd:luser(),
+                     LServer :: ejabberd:lserver()
+                     ) -> boolean() | {error, atom()}.
+does_user_exist(_LUser, _LServer) ->
+    true.
```
Comment (fenek) [inline apps/ejabberd/src/ejabberd_auth_jwt.erl]: Are we sure there are no implicit negative consequences of always returning `true` here?

Comment (michalwski) [inline apps/ejabberd/src/ejabberd_auth_jwt.erl]: There are, but the same we could say about `false` here.

**Pull request merged**